app/src/main/java/org/schabi/newpipe/ui/components/items/ItemList.kt

```diff
@@ -0,0 +1,114 @@
+package org.schabi.newpipe.ui.components.items
+
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListScope
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material3.adaptive.currentWindowAdaptiveInfo
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.rememberNestedScrollInteropConnection
+import androidx.compose.ui.res.stringResource
+import androidx.fragment.app.FragmentActivity
+import androidx.preference.PreferenceManager
+import androidx.window.core.layout.WindowWidthSizeClass
+import my.nanihadesuka.compose.LazyColumnScrollbar
+import org.schabi.newpipe.R
+import org.schabi.newpipe.extractor.InfoItem
+import org.schabi.newpipe.extractor.playlist.PlaylistInfoItem
+import org.schabi.newpipe.extractor.stream.StreamInfoItem
+import org.schabi.newpipe.info_list.ItemViewMode
+import org.schabi.newpipe.ui.components.items.playlist.PlaylistListItem
+import org.schabi.newpipe.ui.components.items.stream.StreamListItem
+import org.schabi.newpipe.util.DependentPreferenceHelper
+import org.schabi.newpipe.util.NavigationHelper
+
+@Composable
+fun ItemList(
+    items: List<InfoItem>,
+    mode: ItemViewMode = determineItemViewMode(),
+    listHeader: LazyListScope.() -> Unit = {}
+) {
+    val context = LocalContext.current
+    val onClick = remember {
+        { item: InfoItem ->
+            val fragmentManager = (context as FragmentActivity).supportFragmentManager
+            if (item is StreamInfoItem) {
+                NavigationHelper.openVideoDetailFragment(
+                    context, fragmentManager, item.serviceId, item.url, item.name, null, false
+                )
+            } else if (item is PlaylistInfoItem) {
+                NavigationHelper.openPlaylistFragment(
+                    fragmentManager, item.serviceId, item.url, item.name
+                )
+            }
+        }
+    }
+
+    // Handle long clicks for stream items
+    // TODO: Adjust the menu display depending on where it was triggered
```

Comment:
I'm a  little new to this code base, so please forgive me if I'm making recommendations that go against the normal process. Typically, I'll recommend that we create a new issue to address "TODO" items since then those can get prioritized and taken care of by anyone else interested in making contributions to the code base.

If this seems like a good idea, please create an issue, and provide a link to it here so that I can review the details, then I'll consider this resolved. Otherwise, if it's common to leave "TODO" comments in the code and address them as they are found by later engineers making changes, then I won't worry about it.

Let me know your thoughts.

Reply:
> Typically, I'll recommend that we create a new issue to address "TODO" items since then those can get prioritized and taken care of by anyone else interested in making contributions to the code base.

Yes, this would be a good idea for normal development. But since we are in the middle of a refactor an this code is going to be looked at again for sure, I wouldn't worry about opening issue and so, as it might just add unneeded overhead. We will need to look at all TODOs before finalizing the refactor.